From ca26404a08a776d0d335ef14a518ab35e255e9fa Mon Sep 17 00:00:00 2001
From: Shreya Singh <shreyacsingh@google.com>
Date: Fri, 11 Apr 2025 08:05:47 -0700
Subject: [PATCH] Skip permission check for SCHEME_ANDROID_RESOURCE

Based on DPMS [documentation](https://source.corp.google.com/h/googleplex-android/platform/superproject/main/+/main:frameworks/base/core/java/android/app/admin/DevicePolicyManager.java;l=1765;drc=33957a5a6a01261255c4d36e132fe2e526875534#:~:text=*%20%3Cp%3EIf%20a%20%3Ccode,%7D%20of%20the%20intent.) FLAG_GRANT_READ_URI_PERMISSION is only required for SCHEME_CONTENT.
Flag: EXEMPT bug fix
Bug:  299928772
Test: Manual
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:5fe27956963c7217f4a46fbaf71e85581c15c75a)
Merged-In: Ie19576c1a2403c62ca94b164736983624ab12b58
Change-Id: Ie19576c1a2403c62ca94b164736983624ab12b58
---
 .../parser/DisclaimersParserImpl.java         | 24 ++++++++++---------
 1 file changed, 13 insertions(+), 11 deletions(-)

diff --git a/src/com/android/managedprovisioning/parser/DisclaimersParserImpl.java b/src/com/android/managedprovisioning/parser/DisclaimersParserImpl.java
index 542bd7ce..9d8bb3b9 100644
--- a/src/com/android/managedprovisioning/parser/DisclaimersParserImpl.java
+++ b/src/com/android/managedprovisioning/parser/DisclaimersParserImpl.java
@@ -127,17 +127,19 @@ public class DisclaimersParserImpl implements DisclaimerParser {
             String errorMessage = "Invalid URI scheme: " + scheme;
             throw new SecurityException(errorMessage);
         }
-        int permissionCheck =
-            mContext.checkUriPermission(
-                uri,
-                Binder.getCallingPid(),
-                Binder.getCallingUid(),
-                Intent.FLAG_GRANT_READ_URI_PERMISSION);
-
-        if (permissionCheck != PackageManager.PERMISSION_GRANTED) {
-            String errorMessage = "Caller does not have permission to access"
-                + " disclaimer URI: " + uri;
-            throw new SecurityException(errorMessage);
+        if (Objects.equals(scheme, SCHEME_CONTENT)) {
+            int permissionCheck =
+                mContext.checkUriPermission(
+                    uri,
+                    Binder.getCallingPid(),
+                    Binder.getCallingUid(),
+                    Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+            if (permissionCheck != PackageManager.PERMISSION_GRANTED) {
+                String errorMessage = "Caller does not have permission to access"
+                    + " disclaimer URI: " + uri;
+                throw new SecurityException(errorMessage);
+            }
         }
     }
 
-- 
2.50.1.351.gcf940e82a1-goog

